src/overlays/actors/ovl_Bg_Haka_Zou/z_bg_haka_zou.c

```diff
@@ -30,35 +56,344 @@ const ActorInit Bg_Haka_Zou_InitVars = {
+            func_800800F8(globalCtx, 3400, 999, &this->dyna.actor, 0);
+        } else if (this->dyna.actor.params == 2) {
+            func_80882E54(this, globalCtx);
+            this->dyna.actor.draw = NULL;
+            this->timer = 1;
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_EXPLOSION);
+            this->actionFunc = func_80883104;
+        } else {
+            func_80882CC4(&this->dyna.actor, globalCtx);
+            this->timer = 1;
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_WALL_BROKEN);
+            this->actionFunc = func_80883104;
+        }
+    } else {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
+
+void func_80883104(BgHakaZou* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (this->timer == 0) {
+        Actor_Kill(&this->dyna.actor);
+    }
+}
+
+void func_80883144(BgHakaZou* this, GlobalContext* globalCtx) {
+    Vec3f effectPos;
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (!(this->timer % 4)) {
+        effectPos.x = Rand_CenteredFloat(200.0f) + (this->dyna.actor.posRot.pos.x - 56.0f);
+        effectPos.y = (Rand_ZeroOne() * 80.0f) + this->dyna.actor.posRot.pos.y;
+        effectPos.z = Rand_CenteredFloat(200.0f) + (this->dyna.actor.posRot.pos.z + 56.0f);
+
+        EffectSsBomb2_SpawnLayered(globalCtx, &effectPos, &sEffectVec, &sEffectVec, 150, 70);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_IT_BOMB_EXPLOSION);
+    }
+
+    if (this->timer == 0) {
+        this->timer = 20;
+        this->actionFunc = func_80883254;
+    }
+}
+
+void func_80883254(BgHakaZou* this, GlobalContext* globalCtx) {
+    f32 moveDist = (Rand_ZeroOne() * 0.5f) + 0.5f;
+
+    Math_StepToF(&this->dyna.actor.posRot.pos.z, this->dyna.actor.initPosRot.pos.z - 80.0f, 2.0f * moveDist);
+
+    if (Math_StepToF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y - 40.0f, moveDist) != 0) {
+        if (this->timer != 0) {
+            this->timer--;
+        }
+
+        if (this->timer == 0) {
+            this->timer = 60;
+            this->dyna.actor.posRot.rot.x = 0x0008;
+            this->actionFunc = func_80883328;
+        }
+    } else {
+        func_808828F4(this, globalCtx);
+    }
+}
+
+void func_80883328(BgHakaZou* this, GlobalContext* globalCtx) {
+    Vec3f effectPos;
+    s32 i;
+    s32 j;
+
+    this->dyna.actor.posRot.rot.x += this->dyna.actor.posRot.rot.x * 0.125f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_808828F4.s")
+    if (Math_ScaledStepToS(&this->dyna.actor.shape.rot.x, -0x4000, this->dyna.actor.posRot.rot.x) != 0) {
+        effectPos.x = this->dyna.actor.posRot.pos.x;
+        effectPos.y = this->dyna.actor.posRot.pos.y;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Zou/func_80882A70.s")
+        for (j = 0; j < 2; effectPos.x -= 112.0f, j++) {
```

Comment:
Can the `effectPos.x -= 112.0f` be moved to the last thing in the loop, by any chance?

Reply:
nope it affects regalloc

Reply:
actually with a bit of changing things around i got it to work